examples/pennylane_run_qaoa_maxcut.py

```diff
@@ -0,0 +1,288 @@
+# To get started, we import PennyLane along with the PennyLane-provided
+# version of NumPy.
+
+
+import pennylane as qml
+from pennylane import numpy as np
+
+
+##############################################################################
+# 2.3 Operators
+# ~~~~~~~~~~~~~
+# We specify the number of qubits (vertices) with ``n_wires`` and
+# compose the unitary operators using the definitions
+# above. :math:`U_B` operators act on each of the wires, while :math:`U_C`
+# operators act on wires whose corresponding vertices are joined by an edge in
+# the graph. We also define the graph using
+# the list ``graph``, which contains the tuples of vertices defining
+# each edge in the graph.
+
+n_wires = 4
+graph = [(0, 1), (0, 3), (1, 2), (2, 3)]
+
+# unitary operator U_B with parameter beta
+def U_B(beta):
+    for wire in range(n_wires):
+        qml.RX(2 * beta, wires=wire)
+
+
+# unitary operator U_C with parameter gamma
+def U_C(gamma):
+    for edge in graph:
+        wire1 = edge[0]
+        wire2 = edge[1]
+        qml.CNOT(wires=[wire1, wire2])
+        qml.RZ(gamma, wires=wire2)
+        qml.CNOT(wires=[wire1, wire2])
+
+
+##############################################################################
+# With some foresight, we notice that we will need a way to sample
+# a measurement of multiple qubits in the computational basis, so we define
+# a Hermitian operator to do this. The eigenvalues of the operator are
+# the joint qubit measurement values in integer form.
+
+
+def comp_basis_measurement(wires):
+    n_wires = len(wires)
+    return qml.Hermitian(np.diag(range(2 ** n_wires)), wires=wires)
+
+
+##############################################################################
+# 2.4 Circuit
+# ~~~~~~~~~~~~~
+# Next, we create a quantum device with 4 qubits.
+
+dev1 = qml.device("default.qubit", wires=n_wires)
+
+##############################################################################
+# We also require a quantum node which will apply the operators :math:`p` times given the
+# angle parameters, and return the expectation value of the observable
+# :math:`\sigma_z^{j}\sigma_z^{k}` to be used in the term of the objective function later on. The
+# argument ``edge`` specifies the chosen edge term in the objective function, :math:`(j,k)`.
+#
+# Once optimized, the same quantum node can be used for sampling an approximately optimal bitstring
+# if executed with the ``edge`` keyword set to None.
+
+pauli_z = [[1, 0], [0, -1]]
+pauli_z_2 = np.kron(pauli_z, pauli_z)
+
+
+@qml.qnode(dev1)
+def circuit(params, edge=None, p=1):
+    # apply hadamards to get n qubit |+> state
+    for wire in range(n_wires):
+        qml.Hadamard(wires=wire)
+    # p instances of unitary operators
+    for i in range(p):
+        U_C(params[0][i])
+        U_B(params[1][i])
+    # during measurement phase we are evaluating
+    # the circuit with optimized parameters
+    if edge is None:
+        return qml.sample(comp_basis_measurement(range(n_wires)), n=1)
```

Comment:
We have just merged a PR that removes the `n` keyword argument from `sample`! This will likely break the tutorial, but we can have a chat about the best way to fix it.